students/ikempcode33/Lesson 1/test_unit.py

```diff
@@ -0,0 +1,147 @@
+"""testing module to evaluate performance of HP Norton"""
+# Adding unit tests for the classes inside inventory_management  folder
+# Isabella Kemp
+
+import sys
+sys.path.append('inventory_management')
+from unittest import TestCase
+from unittest.mock import patch, MagicMock
+#sys.path.append(os.getcwd()+'/inventory_management')
+import inventory_management.market_prices as market_prices
+import inventory_management.main as main
+from inventory_management.inventory_class import Inventory
+from inventory_management.electric_appliances_class import ElectricAppliances
+from inventory_management.furniture_class import Furniture
+
+
+# Class to test inventory class
+class TestInventoryClass(TestCase):
+    def test_inventory(self):
+        inv = {'product_code': '545',
+                    'description': 'pan',
+                    'market_price': 42.0,
+                    'rental_price': 13.0}
+        details = ('545', 'pan', 42.0, 13.0)
+        inv_test = Inventory(*details)
+        self.assertEqual(inv, inv_test.return_as_dictionary())
+
+# Class to test electrical applicances class of inventory management
+class TestElectricAppliancesClass(TestCase):
+    def test_electrical_appliances(self):
+        appliances = {'product_code': '786',
+                      'description': 'phone charger',
+                      'market_price': 400.0,
+                      'rental_price': 130.0,
+                      'brand': 'apple',
+                      'voltage': 2.9}
+        details = ('786', 'phone charger', 400.0, 130.0, 'apple', 2.9)
+        appliances_test = ElectricAppliances(*details)
+        self.assertEqual(appliances, appliances_test.return_as_dictionary())
+
+
+# Class to test the furnitue class
+class TestFurniture(TestCase):
+    def test_furniture(self):
+        furn = {'product_code': '0455',
+                'description': 'couch',
+                'market_prices': 280.0,
+                'rental_price': 30.0,
+                'material': 'leather',
+                'size': 'L'}
+        details = ('0455', 'couch', 280.0, 30.0, 'leather', 'L')
+        furn_test = Furniture(*details)
+        self.assertEqual(furn, furn_test.return_as_dictionary())
+
+
+# Testing Market Prices class
+class TestMarketPrices(TestCase):
+    def test_get_latest_price(self):
+        self.assertEqual(24, market_prices.get_latest_price(24))
+
+
+# Testing main class 
+class TestMain(TestCase):
+    #tests main menu setup
+    def test_main_menu(self):
+        with patch('builtins.input', side_effect='1'):
```

Comment:
You can also group multiple side effects, instead of one per line:

`with patch('builtins.inpu', side_effect=('1', '2', '3'):`